### ISSUE TITLE
Package build order

### ISSUE THREAD

**Issue by hmaarrfk** [Archiconda/build-tools#4]
1. [x] make
2. [x] perl
3. [ ] m4
4. [ ] autoconf
5. [ ] automake
6. [ ] xz

@jjhelmus do you have a good list of the order to follow?
I've already run into many issues with make, perl, and m4, I'm not sure how much I can automate this process.

**Comment by hmaarrfk** [comment]
@jjhelmus do you have a good list of the order to follow?
I've already run into many issues with make, perl, and m4, I'm not sure how much I can automate this process.

**Comment by hmaarrfk** [comment]
When make is found to be necessary, document it here:
https://github.com/conda-forge/docker-images/issues/78

**Comment by jjhelmus** [comment]
The order that I build the bootstrap packages can be found in the [conda4aarch64 repository](https://github.com/jjhelmus/conda4aarch64/blob/master/build_order.md).  I'll update the order as I rebuild the packages in a CentOS 7 container.

I'll be uploading the packages from this build out to the [c4aarch64 channel](https://anaconda.org/c4aarch64/).  GCC 7.3.0 compilers are available there that run on CentOS 7.  This should be preferred over the ones in the aarch64_bootstrap label.

**Comment by hmaarrfk** [comment]
Are you cross compiling? I had major issues with GNUMakefile

**Comment by jjhelmus** [comment]
I'm building inside a Docker container running on a baremetal  aarch64 machine.  Recipe are in the conda4aarch64 repository if you want to compare.

**Comment by jjhelmus** [comment]
m4 recipe is at https://github.com/jjhelmus/conda4aarch64/tree/master/recipes/m4-feedstock/recipe

**Comment by hmaarrfk** [comment]
Is there a way to sync our efforts?

My goal is to follow conda-forge really closely, eventually cleaning up my changes and submitting them back.

Will you be doing a graphical stack?

**Comment by hmaarrfk** [comment]
Yeah, I have the same m4 recipe, it is something weird about either my ubuntu image, or the specific version of crosstools I have :/

Do I need to rebuild all my package with the Centos7 compiler in https://anaconda.org/c4aarch64/repo ? or should it be compatible with the one that worked only on ubuntu 18.04

**Comment by jjhelmus** [comment]
My plan was to build everything needed for conda, conda-build, constructor and anaconda-client.  I want to submit the modification back to conda-forge but working in a single repo initially is easier.

**Comment by jjhelmus** [comment]
> Do I need to rebuild all my package with the Centos7 compiler in https://anaconda.org/c4aarch64/repo ? or should it be compatible with the one that worked only on ubuntu 18.04

The packages existing package should be alright but switching to CentOS 7 base image without compilers or other build tool would be good.  I'll add the image I'm working from to conda4aarch64.

**Comment by hmaarrfk** [comment]
Ok, I'll switch, I just have to change my mods to conda-smithy. Maybe that will fix my m4 issues. If those m4 issues get fixed, then I think we have a good case to seriously start pushing some patches.

I've got the Readme and some basic rerendering down. I'm trying to see if I am going to need to include the arch in the `.ci_support` files. I think it would be prudent to, but will cause the names to explode.

**Comment by jjhelmus** [comment]
https://github.com/jjhelmus/conda4aarch64/tree/master/docker/pkg_builder has the Dockerfile I'm using for builds, I installed an aarch64 miniconda clone that I have locally.

**Comment by hmaarrfk** [comment]
How did you build that aarch64 miniconda? Using the previous bootstrap tags you had? If that is the case, then it is the same as `Archiconda3` I think.

**Comment by jjhelmus** [comment]
Yes, the miniconda I used had packages from the bootstrap, `Archiconda3` should be the same.

**Comment by jjhelmus** [comment]
Did you figure out of shippable has an automated method for registering repositories?  Or is that still a manual step?

**Comment by hmaarrfk** [comment]
So I run that fork repo script with the names of the packages I want, and it does everything from fork, creating the branch, re rendering, quite often, I’ve had to add make to the requirements. 

It can take many repos in parallel, though I haven’t used that much since I’ve hit many packages that require some kind of mod. 

Finally, as you know, this early in the process there is only one thing to build at a time

**Comment by hmaarrfk** [comment]
cc @mingwandroid
